tests/common/AppDelegate.cs

```diff
@@ -48,9 +51,31 @@ public override bool FinishedLaunching (UIApplication application, NSDictionary
 public static class MainClass {
 	static void Main (string [] args)
 	{
+#if __MACCATALYST__
+		NativeLibrary.SetDllImportResolver (typeof (NSObject).Assembly, DllImportResolver);
+		NativeLibrary.SetDllImportResolver (typeof (MainClass).Assembly, DllImportResolver);
+#endif
 #if !__MACOS__
 		UIApplication.Main (args, null, typeof (AppDelegate));
 #endif
 	}
+
+#if __MACCATALYST__
+	// This is a workaround for a temporary issue in the .NET runtime
+	// See https://github.com/xamarin/maccore/issues/2668
+	// The issue is present in .NET 7.0.5, and will likely be fixed in .NET 7.0.6.
```

Comment:
If this is likely to be fixed in 7.0.6, do we have an issue open on it to check in the future to remove this?

Reply:
Filed: https://github.com/xamarin/xamarin-macios/issues/18164